### PR TITLE
K it 148 mailer & contact form

### PR DESCRIPTION
**Pull request by MatteiT**

### CODE REVIEW

File: src/main/java/com/templateproject/api/entity/Contact.java
Comment (emericb) [inline src/main/java/com/templateproject/api/entity/Contact.java:1]: Remove setId for primary

**Review by emericb — CHANGES_REQUESTED** [review]

**Review by emericb — APPROVED** [review]

File: src/main/java/com/templateproject/api/controller/ContactController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    private EmailService emailService;
+
+    public ContactController(ContactService contactService, EmailService emailService) {
+        this.contactService = contactService;
+        this.emailService = emailService;
+    }
+
+    @PostMapping
+    public ResponseEntity<Contact> createContact(@Valid @RequestBody Contact contact) {
+        Contact contactEnregistre = contactService.enregistrerContact(contact);
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/controller/ContactController.java:28]: english variable names

File: src/main/java/com/templateproject/api/service/ContactService.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+@Service
+public class ContactService {
+
+    private final ContactRepository contactRepository;
+
+    public ContactService(ContactRepository contactRepository) {
+        this.contactRepository = contactRepository;
+    }
+
+    public Contact enregistrerContact(Contact contact) {
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/service/ContactService.java:17]: english variable name

File: src/main/java/com/templateproject/api/service/EmailService.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    private final JavaMailSender javaMailSender;
+    private final SpringTemplateEngine templateEngine;
+
+    public EmailService(JavaMailSender javaMailSender, SpringTemplateEngine templateEngine) {
+        this.javaMailSender = javaMailSender;
+        this.templateEngine = templateEngine;
+    }
+
+    public void sendEmail (Contact contact, String subject){
+        // Mail pour l'administrateur
```
Comment (Benja-san) [inline src/main/java/com/templateproject/api/service/EmailService.java:26]: remove dead code

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -31,3 +31,30 @@ spring.jpa.show-sql=true
 # schema will be automatically updated accordingly to java entities found in
 # the project
 spring.jpa.hibernate.ddl-auto=update
+
+spring.mail.default-encoding=UTF-8
+spring.mail.host=smtp.gmail.com
+spring.mail.port=587
+spring.mail.username=BugsSquadInd
```
Comment (Benja-san) [inline src/main/resources/application.properties:38]: put those sensitive datas inside application-dev.properties!

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by emericb — APPROVED** [review]

**Review by emericb — APPROVED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**